From 1bc1a54dfd9c9d448dc671a2bad2f05e0641885c Mon Sep 17 00:00:00 2001
Patch-Source: https://github.com/dotnet/installer/pull/13378
From: Antoine Martin <dev@ayakael.net>
Date: Wed, 9 Mar 2022 08:17:41 +0000
Subject: [PATCH 1/1] arm build

By default, installer does not support arm. This patch introduces all the logics
to parse platform information.

---
 Directory.Build.props | 2 +-
 eng/Build.props       | 2 +-
 2 file changed, 2 insertion(+), 2 deletion(-)

diff --git a/Directory.Build.props b/Directory.Build.props
index c373c9beb..af0f16f88 100644
--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -9,7 +9,7 @@
 
   <PropertyGroup>
     <BuildArchitecture>$([System.Runtime.InteropServices.RuntimeInformation]::ProcessArchitecture.ToString().ToLowerInvariant())</BuildArchitecture>
-    <Architecture Condition="'$(Architecture)' == '' AND '$(BuildArchitecture)' == 'arm64'">$(BuildArchitecture)</Architecture>
+    <Architecture Condition="'$(Architecture)' == '' AND ( '$(BuildArchitecture)' == 'arm64' OR '$(BuildArchitecture)' == 'arm') ">$(BuildArchitecture)</Architecture>
     <Architecture Condition="'$(Architecture)' == '' AND '$(BuildArchitecture)' == 's390x'">$(BuildArchitecture)</Architecture>
     <Architecture Condition="'$(Architecture)' == ''">x64</Architecture>
   </PropertyGroup>
diff --git a/eng/Build.props b/eng/Build.props
index 4df683bab..885669cd9 100644
--- a/eng/Build.props
+++ b/eng/Build.props
@@ -1,7 +1,7 @@
 <Project>
   <ItemGroup Condition=" '$(ArcadeBuildTarball)' != 'true' ">
     <ProjectToBuild Include="$(RepoRoot)Microsoft.DotNet.Cli.sln" />
-    <ProjectToBuild Condition="'$(OS)' == 'Windows_NT' And ('$(Architecture)' == 'x86' Or '$(Architecture)' == 'x64' Or '$(Architecture)' == 'arm64')" 
+    <ProjectToBuild Condition="'$(OS)' == 'Windows_NT' And ('$(Architecture)' == 'x86' Or '$(Architecture)' == 'x64' Or '$(Architecture)' == 'arm64' Or '$(Architecture)' == 'arm')" 
                     Include="$(RepoRoot)eng\version.csproj;
                              $(RepoRoot)eng\native.proj" />
   </ItemGroup>
-- 
2.35.1

